control_panel_api/aws.py

```diff
@@ -0,0 +1,74 @@
+import json
+
+import boto3
+
+
+def create_bucket(name, region, **kwargs):
+    boto3.client('s3').create_bucket(
+        Bucket=name,
+        ACL=kwargs.get('acl', 'private'),
+        CreateBucketConfiguration={'LocationConstraint': region},
+    )
+
+
+def delete_bucket(name):
+    boto3.client('s3').delete_bucket(Bucket=name)
+
+
+def put_bucket_logging(name, target_bucket, target_prefix):
+    boto3.client('s3').put_bucket_logging(
+        Bucket=name,
+        BucketLoggingStatus={
+            'LoggingEnabled': {
+                'TargetBucket': target_bucket,
+                'TargetPrefix': target_prefix
+            }
+        }
+    )
+
+
+def create_policy(name, policy_document):
+    boto3.client("iam").create_policy(
+        PolicyName=name,
+        PolicyDocument=json.dumps(policy_document)
+    )
+
+
+def delete_policy(policy_arn):
+    boto3.client("iam").delete_policy(PolicyArn=policy_arn)
+
+
+def detach_policy_from_entities(policy_arn):
+    """Get all entities to which policy is attached
+    See: http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.list_entities_for_policy"""
+    entities = boto3.client("iam").list_entities_for_policy(PolicyArn=policy_arn)
+
+    for role in entities["PolicyRoles"]:
+        detach_policy_from_role(policy_arn, role["RoleName"])
+
+    for group in entities["PolicyGroups"]:
+        detach_policy_from_group(policy_arn, group["GroupName"])
+
+    for user in entities["PolicyUsers"]:
+        detach_policy_from_user(policy_arn, user["UserName"])
+
+
+def detach_policy_from_role(policy_arn, role_name):
```

Comment:
All these `detach_policy_from_*` functions could probably also be "private" as they're only used internally at this stage.

Reply:
Yes they are used just internally but they are still fundamentally aws api calls so I would keep them open for any possible future usage, and not make it confusing why they are designated private or public

Reply:
Mmmm, we can keep them public if you think it's better 👍 

In my head this is still something the outside code doesn't care because it's only necessary to delete an IAM role (at this stage).